termuic/termuiConsole.go

```diff
@@ -0,0 +1,276 @@
+
+		tc.mutLogLineWrite.Lock()
+		if len(tc.logLines) >= tc.numLogLines {
+			tc.logLines = tc.logLines[1:tc.numLogLines]
+		}
+		tc.mutLogLineWrite.Unlock()
+
+		logLine = strings.Replace(logLine, "\n", "", len(logLine))
+
+		logLine = strings.Replace(logLine, "\r", "", len(logLine))
+
+		if logLine != "" {
+			tc.mutLogLineWrite.Lock()
+			tc.logLines = append(tc.logLines, logLine)
+			tc.mutLogLineWrite.Unlock()
+		}
+	}()
+
+	return len(p), nil
+}
+
+// InitMetricsMap will init the map of prometheus metrics
+func (tc *TermuiConsole) initMetricsMap() {
+	tc.TermuiConsoleMetrics = sync.Map{}
+
+	tc.TermuiConsoleMetrics.Store(core.MetricCurrentRound, 0)
+	tc.TermuiConsoleMetrics.Store(core.MetricIsSyncing, 0)
+	tc.TermuiConsoleMetrics.Store(core.MetricNonce, 0)
+	tc.TermuiConsoleMetrics.Store(core.MetricNumConnectedPeers, 0)
+	tc.TermuiConsoleMetrics.Store(core.MetricSynchronizedRound, 0)
+
+	tc.TermuiConsoleMetrics.Store(core.MetricPublicKey, "")
+	tc.TermuiConsoleMetrics.Store(core.MetricShardId, 0)
+	tc.TermuiConsoleMetrics.Store(core.MetricTxPoolLoad, 0)
+
+	tc.TermuiConsoleMetrics.Store(core.MetricCountConsensus, 0)
+	tc.TermuiConsoleMetrics.Store(core.MetricCountLeader, 0)
+	tc.TermuiConsoleMetrics.Store(core.MetricCountAcceptedBlocks, 0)
+}
+
+// SetInt64Value method - will update the value for a key
+func (tc *TermuiConsole) SetInt64Value(key string, value int64) {
+	if _, ok := tc.TermuiConsoleMetrics.Load(key); ok {
+		tc.TermuiConsoleMetrics.Store(key, int(value))
+	}
+}
+
+// SetUInt64Value method - will update the value for a key
+func (tc *TermuiConsole) SetUInt64Value(key string, value uint64) {
+	if _, ok := tc.TermuiConsoleMetrics.Load(key); ok {
+		tc.TermuiConsoleMetrics.Store(key, int(value))
+	}
+}
+
+// SetStringValue method - will update the value of a key
+func (tc *TermuiConsole) SetStringValue(key string, value string) {
+	if _, ok := tc.TermuiConsoleMetrics.Load(key); ok {
+		tc.TermuiConsoleMetrics.Store(key, value)
+	}
+}
+
+// Increment - will increment the value of a key
+func (tc *TermuiConsole) Increment(key string) {
+	if keyValueI, ok := tc.TermuiConsoleMetrics.Load(key); ok {
+
+		keyValue := keyValueI.(int)
+		keyValue++
+		tc.TermuiConsoleMetrics.Store(key, keyValue)
+	}
+}
+
+// Decrement method - will decrement the value for a key
+func (tc *TermuiConsole) Decrement(key string) {
+	return
+}
+
+// Close method - won't do anything
+func (tc *TermuiConsole) Close() {
+	return
+}
+
+// Start method - will start termui console
+func (tc *TermuiConsole) Start() {
+	go func() {
+		if err := ui.Init(); err != nil {
+
+		}
+		defer ui.Close()
+
+		tc.eventLoop()
+	}()
+
+}
+
+func (tc *TermuiConsole) eventLoop() {
+
+	tc.termuiConsoleWidgets = NewtermuiConsoleGrid()
+
+	time.Sleep(1 * time.Second)
+
+	grid := tc.configConsoleWithNormalTermuiWidgets()
+
+	uiEvents := ui.PollEvents()
+	// handles kill signal sent to gotop
+	sigTerm := make(chan os.Signal, 2)
+	signal.Notify(sigTerm, os.Interrupt, syscall.SIGTERM)
```

Comment:
Doesn't this conflict with the main's signal notifier?

Reply:
This signal notifier does not conflict with the main signal notifier because i need this signal when termui console must be closed by goland and the console needs to be cleared. Main's signal notifier will be notified when user press "Ctrl+c" a signal will be injected when event "Ctrl+c" will come.